vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleOverviewManager.java

```diff
@@ -0,0 +1,84 @@
+import edu.asu.diging.vspace.core.services.impl.model.SequenceOverview;
+import edu.asu.diging.vspace.core.services.impl.model.SlideOverview;
+
+@Service
+public class ModuleOverviewManager implements IModuleOverviewManager {
+    
+    @Autowired
+    private IModuleManager moduleManager;
+    
+    public ModuleOverview getModuleOverview(String moduleId) {
+        IModule module = moduleManager.getModule(moduleId);
+        ISequence startSequence = module.getStartSequence();
+        
+        List<ISequence> sequences = moduleManager.getModuleSequences(moduleId);
+        
+        SequenceOverview startSequenceNode = createSequenceOverviewNode(startSequence);
+        List<SequenceOverview> otherSequences = new ArrayList<SequenceOverview>();
+        for(ISequence sequence : sequences) {
+            
+            if(sequence != startSequence) {
+                otherSequences.add(createSequenceOverviewNode(sequence));
+            }
+        }
+        
+        ModuleOverview moduleOverview = new ModuleOverview();
+        moduleOverview.setStartSequence(startSequenceNode);
+        moduleOverview.setOtherSequences(otherSequences);
+        moduleOverview.setStartSlides(startSequence.getSlides());
+        return moduleOverview;
+    }
+    
+    /**
+     * This method is used to fetch all Sequences which belong to a module and 
+     * convert this into a SequenceOverview node. The SequenceOverviewNode is added to 
+     * ModuleOverview
+     * @return ModuleOverview which contains the module and the list of sequences and its slides
+     */   
+    private SequenceOverview createSequenceOverviewNode(ISequence sequence) {
+        
+        SequenceOverview sequenceOverview = new SequenceOverview();
+        sequenceOverview.setName(sequence.getName());
+        sequenceOverview.setId(sequence.getId());
+        List<SlideOverview> slideOverviews = createSlideOverviewNode(sequence.getSlides());
+        sequenceOverview.setSlideOverviews(slideOverviews);
+        return sequenceOverview;  
+    }
+    
+    private List<SlideOverview> createSlideOverviewNode(List<ISlide> slides){
+        List<SlideOverview> slideOverviews = new ArrayList<SlideOverview>();
+        for(ISlide slide : slides) {
+            SlideOverview slideOverview = new SlideOverview(); 
+            slideOverview.setId(slide.getId());
+            slideOverview.setName(slide.getName());
+            if(slide instanceof BranchingPoint ) {
+                slideOverview.setBranchingPoint(true);
+                List<IChoice> sequenceChoices = ((BranchingPoint)slide).getChoices();
+                List<String> slideOverviewSequenceChoices = new ArrayList<String>();
+                sequenceChoices.stream().forEach(sequenceChoice -> slideOverviewSequenceChoices.add(
+                        sequenceChoice.getSequence().getName()));
+                slideOverview.setSequenceIds(slideOverviewSequenceChoices);
```

Comment:
I'm so confused, the variable is called `slideOverviewSequenceChoices`, it is filled with the names of the linked sequences, but it is set as "sequenceIds" in the slideOverview? what is it ids, names, or choices?